File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.22-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_22() {
    let column = 10;
    let row = 5;

    for (let y = 1; y < row + 1; y++) {
        let counter = 1;

        for (let x = column; x >= 0; x--) {
            if (x > column / 2) {
                let x_element = x - column / 2;

                if (x_element < y) {
                    log(counter);
                    counter = counter + 1;
                } else {
                    log(' ');
                }
            } else if (x <= column / 2) {
                let y_element = column / 2 - x;

                if (y_element < y) {
                    log(counter);
                    counter = counter - 1;
                } else {
                    log(' ');
                }
            }
        }
        counter = 1;
        console.log();
    }
}

pattern_22();

File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.22-pattern.py
from utils.clear import clear

def pattern_22():
    row = 5
    column = 10

    counter = 0

    for i in range(row, 0, -1):
        for j in range(1, column + 1):
            if j <= column / 2:
                if j >= i:
                    counter = counter + 1
                    print(counter, end="")
                elif j < i:
                    print(" ",end="")
            elif j > column / 2:
                key = column - j
                if key >= i:
                    counter = counter - 1
                    print(counter, end="")
                elif key < i:
                    print(" ",end="")

        counter = 0
        print()

clear()
pattern_22()
